src/game/GameManager.cpp

```diff
@@ -692,13 +814,40 @@ int32_t GameManager::createWeaponStore(const float x, const float y) {
  * creates a square area of DropPoints
  */
 void GameManager::createDropZone(const float x, const float y, const int num) {
+    dropZoneCoord.first = x;
+    dropZoneCoord.second = y;
     for (int i = 0; i < num; i++) {
         for (int j = 0; j < num; j++) {
             createDropPoint(x + (DROP_POINT_SPACE * i), y + (DROP_POINT_SPACE * j));
         }
     }
 }
 
+int32_t GameManager::createBarricadeDrop(const float x, const float y){
+    const int32_t id = generateID();
+
+    SDL_Rect barricadeDropRect = {static_cast<int>(x),static_cast<int>(y), DEFAULT_SIZE, DEFAULT_SIZE};
+    SDL_Rect pickRect = {static_cast<int>(x),static_cast<int>(y), DEFAULT_SIZE, DEFAULT_SIZE};
+
+    barricadeDropManager.emplace(id, BarricadeDrop(id, barricadeDropRect, pickRect))->second.setPosition(x,y);
+    getBarricadeDrop(id).setSrcRect(WALL_SRC_X, WALL_SRC_Y, WALL_SRC_W, WALL_SRC_H);
+    return id;
+}
+
+BarricadeDrop& GameManager::getBarricadeDrop(const int32_t id) {
+
+    const auto& bd = barricadeDropManager[id];
+    return bd.first;
+    assert(bd.second);
+}
+
+void GameManager::deleteBarricadeDrop(int32_t id){
+        barricadeDropManager.erase(id);
+    /*#ifdef SERVER
+        saveDeletion({UDPHeaders::BARRICADEDROP, id});
+    #endif*/
```

Comment:
Barricade Drops arent on the server but i placed it there cause i figured it would be implemented similar as weapondrops were but commented it out cause it would break the server